### ISSUE TITLE
The `.json()` function call is stuck on promise "pending" in middleware

### ISSUE THREAD

**Issue by maskeynihal** [maskeynihal/laravel-with-experimental-nextjs#1]
Steps to reproduce: 
1. Complete the installation of both [backend](https://github.com/maskeynihal/laravel-with-experimental-nextjs#installation) and [frontend](https://github.com/maskeynihal/laravel-with-experimental-nextjs#installation-1)
2. Go to `/login`
3. Click on `Login` button _email and password are predefined_
4. Redirected on success to `/dashboard` and logged in user is shown. 
5. The whole process is completed. _View the console_
<img width="542" alt="image" src="https://user-images.githubusercontent.com/26411488/235414921-0858db90-23fb-43be-b0ad-106b67e2b0a2.png">

6. Now go to `/middleware/dashboard`, the page is stuck
7. The process is stuck on `.json()`  ie `Converting request to json`. 
<img width="486" alt="image" src="https://user-images.githubusercontent.com/26411488/235415040-8de321e6-88e4-49b2-9a31-46d31bca74a0.png">

----
TLDR; 
⚠️ The `.json()` function of [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) is stuck in [middleware](https://nextjs.org/docs/advanced-features/middleware) but working fine in server components.

**Comment by maskeynihal** [comment]
Issue mentioned: 
1. https://github.com/vercel/next.js/discussions/41745#discussioncomment-5769476
